<!-- {about} --><a name="about"></a>
<p><a name="about" id="about"></a></p>
<h2 id="about" class="first">About the <em>Fusarium graminearum</em><!--nbsp--> <!--nbsp-->genome</h2>
<p><em style="line-height: 1.5em;">Fusarium graminearum</em>, also known by<!--nbsp--> <!--nbsp--><em style="line-height: 19.2000007629395px;">Gibberella zeae</em><!--nbsp--> <!--nbsp-->(by which name it was previously known in Ensembl Fungi),<span style="line-height: 1.5em;"><!--nbsp--> <!--nbsp-->is one of the causal agents of head blight disease in wheat and barley. This devastating pathogen and other head blight causes important losses in crops worldwide.</span></p>
<p>Picture credit: <a href="http://en.wikipedia.org/wiki/File:F.graminearum.JPG">Fk</a></p>
<!-- {about} -->
<!-- {annotation} --><a name="annotation"></a>
<p><a name="annotation" id="annotation"></a></p>
<h3 id="annotation">Assembly and Annotation</h3>
<p>This version of the genome sequence of the PH-1 strain has been provided by Rothamsted Research and the Centre for Genome Analysis and has been submitted to the International Nucleotide Sequence Database Consortium (INSDC) as<!--nbsp--> <!--nbsp--><a href="http://www.ebi.ac.uk/ena/data/view/PRJEB5475">PRJEB5475</a>. The dataset was produced by combining the draft version of the genome previously generated by the Broad Institute using Sanger sequencing and manually annotated<span style="line-height: 1.5em;"><!--nbsp--> <!--nbsp-->by the Fungal and Microbial genomics group at the Munich Information Services for Protein Sequences<!--nbsp--> <!--nbsp--></span><span style="line-height: 1.5em;">(submitted to INSDC as<!--nbsp--> <!--nbsp--></span><a href="http://www.ebi.ac.uk/ena/data/view/GCA_000240135" style="line-height: 1.5em;">GCA_000240135</a><span style="line-height: 1.5em;">), with additional whole shotgun data. In addition to filling gaps in the previous assembly, the new version has: identified a mis-assembly in chromosome 1 at the 5’ end, and extended the AT rich regions, which include the identification for the first time of telomere and centromere sequences. This has increased the genomic sequence by 484,290 bp. Four previously unplaced supercontigs have been located within the new assembly <!--nbsp--> <!--nbsp-->or within the mitochrondrial sequence. The remaining unlocated supercontigs have been identified as repeat sequences, with all but one matched in part to the repeat sequence at the 3’ end of chromosome 4 (which has not been counted in the estimate of the total genome length). In addition to the chromosomal updates, a revised mitochondrial sequence has also been produced.</span></p>
<p class="space-below">Non coding RNA genes have been annotated using tRNAScan-SE (Lowe, T.M. and Eddy, S.R. 1997), RFAM (Griffiths-Jones et al 2005), and RNAmmer (Lagesen K.,et al 2007); additional analysis tools have also been applied.</p>
<!-- {annotation} -->
<!-- {variation} --><a name="variation"></a>
<p><a name="variation" id="variation"></a></p>
<h3 id="variation">Variation</h3>
<p>Variation<!--nbsp--> <!--nbsp-->data has been obtained by comparing the sequenced strain PH-1 with Ë0.4-fold coverage of whole-genome shotgun sequence from a second strain of <i>F.<!--nbsp--> <!--nbsp-->graminearum</i><span style="line-height: 1.5em;">, GZ3639. The identified 10,308 single-nucleotide polymorphisms (SNPs) between the two strains and mapped SNP positions were directly imported from </span><a href="http://www.broadinstitute.org" style="line-height: 1.5em;">Broad Institute</a><span style="line-height: 1.5em;">. This data has also been submitted to dbSNP and is expected to be part of dbSNP 135.</span></p>
<!-- {variation} -->
<!-- {other} --><a name="other"></a>
<h3>Other Data</h3>
<p>'Expressed Sequence Tags' (ESTs) from dbEST are aligned to the genome using Exonerate.</p>
<!-- {other} -->
<h3>References</h3><ol><li><a id="ref-01"></a><a href="http://europepmc.org/abstract/MED/12620255">Analysis of expressed sequence tags from Gibberella zeae (anamorph Fusarium graminearum)</a>.<br>Trail F, Xu JR, San Miguel P, Halgren RG, Kistler HC. 2003. Fungal Genet. Biol.. 38:187-197.</li>
<li><a id="ref-02"></a><a href="http://europepmc.org/abstract/MED/20237561">Comparative genomics reveals mobile pathogenicity chromosomes in Fusarium</a>.<br>Ma LJ, van der Does HC, Borkovich KA, Coleman JJ, Daboussi MJ, Di Pietro A, Dufresne M, Freitag M, Grabherr M, Henrissat B et al. 2010. Nature. 464:367-373.</li>
<li><a id="ref-03"></a><a href="http://europepmc.org/abstract/MED/17823352">The Fusarium graminearum genome reveals a link between localized polymorphism and pathogen specialization</a>.<br>Cuomo CA, Gldener U, Xu JR, Trail F, Turgeon BG, Di Pietro A, Walton JD, Ma LJ, Baker SE, Rep M et al. 2007. Science. 317:1400-1402.</li></ol>